src/main/java/net/snowflake/ingest/streaming/internal/StorageManager.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+import java.util.Optional;
+
+/**
+ * Interface to manage {@link StreamingIngestStorage} for {@link FlushService}
+ *
+ * @param <T> The type of chunk data
+ * @param <TLocation> the type of location that's being managed (internal stage / external volume)
+ */
+interface StorageManager<T, TLocation> {
```

Comment:
nit: IStorageManager, since its an interface? (similarly for all other interfaces?)
I don't think this is something we've really followed consistently everywhere else, but check with Toby on whether this is something we want to start.

Reply:
yeah, IStorageManager is better